File: Impostor/MainGameViewController.m
//
//  MainGameViewController.m
//  Impostor
//
//  Created by William Entriken on 12/14/13.
//  Copyright (c) 2013 William Entriken. All rights reserved.
//

#import "Impostor-Swift.h"
#import "MainGameViewController.h"
#import "SecretWordViewController.h"

@interface MainGameViewController ()
@property ImpostorGameModel *game;
@end

@implementation MainGameViewController

- (void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];

    switch ([[ImpostorGameModel game] gameStatus]) {
        case GameStatusShowSecretWords:
            [self performSegueWithIdentifier:@"secretWord" sender:self];
            break;
        case GameStatusTheImpostorRemains:
            [self performSegueWithIdentifier:@"elimination" sender:self];
            break;
        case GameStatusTheImpostorWasDefeated:
        case GameStatusTheImpostorWon:
            [self performSegueWithIdentifier:@"results" sender:self];
            break;
    }
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if ([segue.destinationViewController isKindOfClass:[SecretWordViewController class]]) {
        SecretWordViewController *controller = (SecretWordViewController *)segue.destinationViewController;
        controller.playerNumber = 0;
    }
}


@end

File: Impostor/MainGameViewController.swift
//
//  MainGameViewController.swift
//  Impostor
//
//  Created by William Entriken on 4/24/16.
//  Copyright © 2016 William Entriken. All rights reserved.
//

import UIKit

class MainGameViewController: UIViewController {
    override func viewDidAppear(_ animated: Bool) {        
        super.viewDidAppear(animated)
        switch ImpostorGameModel.game.gameStatus {
        case .showSecretWord:
            self.performSegue(withIdentifier: "secretWord", sender: self)
        case .theImpostorRemains:
            self.performSegue(withIdentifier: "elimination", sender: self)
        case .theImpostorWasDefeated, .theImpostorWon:
            self.performSegue(withIdentifier: "results", sender: self)
        }
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        switch ImpostorGameModel.game.gameStatus {
        case .showSecretWord(let playerNumber):
            let secretWordViewController = segue.destination as! SecretWordViewController
            secretWordViewController.playerNumber = playerNumber
        default:
            break
        }
    }
}
